code/modules/projectiles/guns/energy.dm

```diff
@@ -89,13 +96,104 @@ GLOBAL_LIST_INIT(registered_cyborg_weapons, list())
 					return suit.cell
 	return null
 
+/obj/item/weapon/gun/energy/special_check(var/mob/user)
+
+	if(!..())
+		return
+	
+	if(hatch_open)
+		to_chat(user, "<span class='warning'>[src] has its cell cover still open!</span>")
+		return 0
+	return 1
+
+//To load a new power cell into the energy gun, if item A is a cell type and the gun is not self-recharging
+/obj/item/weapon/gun/energy/proc/load_ammo(var/obj/item/A, mob/user)
+	if(self_recharge)
+		return
+	//only let's you load in power cells
+	if(istype(A, /obj/item/weapon/cell))
+		. = TRUE
+		var/obj/item/weapon/cell/AM = A
```

Comment:
```suggestion
/obj/item/weapon/gun/energy/proc/load_ammo(var/obj/item/weapon/cell/AM, mob/user)
	if(self_recharge)
		return
	//only let's you load in power cells
	if(istype(AM))
		. = TRUE

```
Might be me being overly picky and sorry if this is out of line, but this reassignment seems redundant rather than just checking the type defined?

Reply:
Thanks that makes sense and tested it, I implemented this in the latest commit.